tests/rptest/tests/datalake/databricks_test.py

```diff
@@ -230,6 +231,97 @@ class TestCase:
                     f"Expected DBX schema {tc.dbx_schema} but got {actual_schema}"
 
+    @cluster(num_nodes=2)
+    @matrix(cloud_storage_type=supported_storage_types())
+    def test_e2e_with_partition_evolution(self, cloud_storage_type):
+        # TODO: Move this in the matrix decorator. Somehow.
+        if not DatabricksContext.available(self.test_context):
+            self.logger.warning(
+                "Skipping test because Databricks context is not available")
+            cleanup_on_early_exit(self)
+            return
+
+        if get_cloud_provider() != "aws":
+            self.logger.warning(
+                f"Skipping test because it is only supported on AWS, but the current cloud provider is {get_cloud_provider()}"
+            )
+            cleanup_on_early_exit(self)
+            return
+
+        with DatalakeServices(
+                self.test_context,
+                redpanda=self.redpanda,
+                include_query_engines=[QueryEngineType.DATABRICKS_SQL],
+                catalog_type=CatalogType.DATABRICKS_UNITY) as dl:
+
+            dl.create_iceberg_enabled_topic(
+                self.topic_name,
+                config={"redpanda.iceberg.partition.spec": "()"})
+
+            produced_total = 0
+
+            def produce_some_and_wait_for_translation():
+                num_msgs = 100
+                nonlocal produced_total
+
+                dl.produce_to_topic(self.topic_name, 1024, num_msgs)
+                produced_total += num_msgs
+
+                dl.wait_for_translation(self.topic_name,
+                                        msg_count=produced_total,
+                                        timeout=60)
+
+            self.logger.info(
+                "Producing data to the topic with no partitioning")
+            produce_some_and_wait_for_translation()
+
+            rpk = RpkTool(self.redpanda)
+
+            self.logger.info(
+                "Producing data to the topic with partitioning by hour and bucket"
+            )
+            rpk.alter_topic_config(
+                self.topic_name, "redpanda.iceberg.partition.spec",
+                "(hour(redpanda.timestamp), bucket(4, redpanda.offset))")
+            produce_some_and_wait_for_translation()
+
+            self.logger.info(
+                "Producing data to the topic with partitioning by bucket only")
+            rpk.alter_topic_config(self.topic_name,
+                                   "redpanda.iceberg.partition.spec",
+                                   "(bucket(4, redpanda.offset))")
+
+            produce_some_and_wait_for_translation()
+
+            def fetch_dbx_schema():
+                table_name = f"redpanda.{self.topic_name}"
+                return dl.query_engine(
+                    QueryEngineType.DATABRICKS_SQL).make_client().cursor(
+                    ).execute(f"describe {table_name}").fetchall()
+
+            actual_schema = fetch_dbx_schema()
```

Comment:
nit: hmm why not use the existing `fetch_dbx_schema()`?

Reply:
That was the intention. Bad rebasing on my end